eng/pipelines/maintenance-packages.yml

```diff
@@ -79,40 +79,9 @@ extends:
               fetchDepth: 0
               clean: true
             steps:
-            - script: eng\common\cibuild.cmd -configuration Release -prepareMachine $(_AdditionalBuildArgs) $(_InternalBuildArgs)
+            - script: eng\common\cibuild.cmd -configuration Release -prepareMachine $(_AdditionalBuildArgs) $(_InternalBuildArgs) /p:IsPackable=true
```

Comment:
Shouldn't this set `IsSourceProjectPackable` and not `IsPackable`?

Reply:
OK I see that this pipeline now only applies to the official build. Do we need to set IsPackable in the official build?

Reply:
Hmmm... `eng/common/cibuild.cmd` runs this internally:
https://github.com/dotnet/maintenance-packages/blob/7b9d2b12325acaef509187a37b769983a74cc69f/eng/common/CIBuild.cmd#L1-L2

Note it already includes the `-pack` argument. That means, I'd like to assume, that I need to remove the `IsPackable=true` argument, and ensure packing is done with just `-pack`. I also need to confirm `-publish` is respected.

Reply:
This is making me think that the maintenance-packages-public.yml PR builds should _not_ call cibuild.cmd directly, because we don't want those to pack, publish or sign (do we want to sign?). Instead, I think we should directly call:

```
build.ps1 -restore -build -test -ci -repareMachine -configuration $(_BuildConfig) $(_AdditionalBuildArgs) $(_InternalBuildArgs)
```

Thoughts @ViktorHofer @ericstj?

Reply:
I also noticed that the official build job, when run internally, passes `/p:Test=false`, even though we are running the eng/common/cibuild.cmd script which passes -test by default. 

https://github.com/dotnet/maintenance-packages/blob/63a086bdf5d4dc194e618863619707a271cecb35/eng/pipelines/maintenance-packages.yml#L16-L25

Reply:
> This is making me think that the maintenance-packages-public.yml PR builds should not call cibuild.cmd directly, because we don't want those to pack, publish or sign (do we want to sign?)

cibuild.cmd is the right entry point for both public and internal. When running on public, signing does a dry run and publishing is skipped automatically. In addition, it makes sense to always pack in all legs. The Pack target only runs when a project sets IsPackable to true.

We just need one special leg that sets IsPackable globally to true for PR validation.

Reply:
Alright, thanks for the explanation.

Reply:
Going back to your question:

> Why do we need to set IsPackable=true in the official build?

If IsPackable=true is what actually turns on all the packages globally, and cibuild.cmd is already including the -pack, -sign and -publish arguments, don't we want to actually generate the packages?

Reply:
My bad, we don't want that. We only build the projects that have their property set to true. Ignore me :)